src/main/java/org/example/App.java

```diff
@@ -54,24 +56,54 @@ public static void main( String[] args ) {
 
         while (!exit) {
             // Get start and end coordinates from user input
-            System.out.print("Enter the ID of the start point (or 'exit' to quit): ");
-            String startXInput = scan.next();
-
-            // Check if user wants to exit
-            if (startXInput.equalsIgnoreCase("exit")) {
-                exit = true;
-                continue;
+            System.out.println("\nPlease set your location and destination.");
+            System.out.println("Enter 'exit' if you would like to quit.\n");
+
+            String startXInput = null;
+            boolean startIdInvalid = true;
+            while (startIdInvalid) {
+                // Check if user wants to exit
+                System.out.print("Enter the ID of the start point: ");
+                startXInput = scan.next();
+
+                if (startXInput.equalsIgnoreCase("exit")) {
+                    exit = true;
+                    break;
+                } else if (!NumberUtils.isNumber(startXInput)) {
+                    System.out.println("ID must be numeric or enter 'exit' to quit.");
+                } else
+                    startIdInvalid = false;
             }
 
+            if (exit)
+                break;
+
+
             int startXC = Integer.parseInt(startXInput);
 
-            System.out.print("Enter the ID of the end point: ");
-            int endYC = scan.nextInt();
+            String endYInput = null;
+            boolean endIdInvalid = true;
+            while (endIdInvalid) {
+                // Check if user wants to exit
+                System.out.print("Enter the ID of the end point: ");
+                endYInput = scan.next();
+
+                if (endYInput.equalsIgnoreCase("exit")) {
+                    exit = true;
+                    break;
+                } else if (!NumberUtils.isNumber(endYInput)) {
+                    System.out.println("ID must be numeric.");
+                } else
+                    endIdInvalid = false;
+            }
+            if (exit)
+                break;
+
+            int endYC = Integer.parseInt(endYInput);
 
             Point starts = null;
             Point ends = null;
-//            Point starts = new Point(1);
-//            Point ends = new Point(2);
+
 
```

Comment:
remove space between ( and the opening " in (    " + endYC + ") in lines 131, 138, 142, 147, 172 (it didn't let me comment in those lines))